src/Gui/Window.h

```diff
@@ -46,6 +46,19 @@ class GuiExport WindowParameter : public ParameterGrp::ObserverType
   /// return the parameter group of this window
   ParameterGrp::handle getWindowParameter();
 
+  /// connect slot to ParameterManager signal
+  template<typename S, typename T>
+  void setSlotParamChanged(S slot, T* obsPtr)
+  {
+    namespace bp = boost::placeholders;
+    if (_handle->Manager()) {
+        connParamChanged = _handle->Manager()->signalParamChanged.connect(
+            boost::bind(slot, obsPtr, bp::_1 ,bp::_2 ,bp::_3 ,bp::_4));
+    }
+  }
+
+  boost::signals2::connection connParamChanged;

```

Comment:
Does this have to be public?

Reply:
Not at all. It should be private.
Do you agree with move `setSlotParamChanged` and `setGroupName` to protected?

Reply:
Definitely on `setSlotParamChanged`. I don't feel strongly about changing `setGroupName`. Is there a compelling reason to put the function definition in the header instead of the cpp file?

Reply:
I put it there because it is a template. What alternative do you suggest?

Reply:
lol, yep, that's a "compelling reason"! Sorry for the noise.